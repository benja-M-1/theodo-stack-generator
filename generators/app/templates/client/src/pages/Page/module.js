/**
 * Following the duck pattern, actions, constants and reducers are in the same file called module.js
 * 
 * See: https://github.com/erikras/ducks-modular-redux
 * 
 */

 /**
 * Constants should be scoped to their module: use the string Page/ADD_ITEM instead of ADD_ITEM
 */
const ADD_ITEM = 'Page/ADD_ITEM';

export function addItem(item) {
  return {
    type: ADD_ITEM,
    payload: item,
  };
}

const initialState = {
  list: [
    { id: 1, label: 'item1' },
    { id: 2, label: 'item2' },
  ],
};

/**
 * Following the duck pattern, the module.js file should export a reducer as a default function
 */
export default function reducer(state = initialState, action) {
  switch (action.type) {
    case ADD_ITEM:
      return {
        ...state,
        list: [
          ...state.list,
          { id: state.list.length + 1, label: action.payload },
        ],
      };
    default:
      return state;
  }
}
