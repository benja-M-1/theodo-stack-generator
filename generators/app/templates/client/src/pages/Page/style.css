.container {
  display: flex;
  justify-content: center;
}

.itemContainer {
  margin-top: 30px;
  color: grey;
}
