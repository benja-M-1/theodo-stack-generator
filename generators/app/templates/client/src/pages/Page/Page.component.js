import React, { Component } from 'react';
import { PropTypes } from 'prop-types';
import { Link } from 'react-router';
import { FormattedMessage } from 'react-intl';
import './style.css';

export default class Page extends Component {
  static propTypes = {
    intl: PropTypes.object.isRequired, // eslint-disable-line
    items: PropTypes.arrayOf(PropTypes.shape({
      id: PropTypes.number,
      label: PropTypes.string,
    })),
    addItem: PropTypes.func.isRequired,
  };

  render() {
    const { formatMessage } = this.props.intl;
    return (
      <div className="page-container">
        <Link to="/">
          <FormattedMessage id="page.back" />
        </Link>
        <p>{formatMessage({ id: 'page.api-to-translate-example' })}</p>
        <button onClick={() => this.props.addItem('new item')}>
          <FormattedMessage id="home.click-me" />
        </button>
        <div className="itemContainer">
          { this.props.items.map(item => (
            <div key={item.id} >{item.label}</div>
          ))}
        </div>
      </div>
    );
  }
}
