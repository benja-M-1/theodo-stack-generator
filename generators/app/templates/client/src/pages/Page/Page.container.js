import { connect } from 'react-redux';
import { injectIntl } from 'react-intl';
import { addItem } from './module';

import Page from './Page.component';

const PageWithIntl = injectIntl(Page);

function mapStateToProps(state) {
  return {
    items: state.page.list,
  };
}

function mapDispatchToProps(dispatch) {
  return {
    addItem(item) {
      dispatch(addItem(item));
    },
  };
}

export default connect(
  mapStateToProps,
  mapDispatchToProps,
)(PageWithIntl);
