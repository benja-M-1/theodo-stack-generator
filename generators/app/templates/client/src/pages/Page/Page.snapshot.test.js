import React from 'react';
import Page from './Page.component';
import createComponentWithIntl from '../../utils/i18n/create-component-with-intl';

describe('The page component', () => {
  it('should renders correctly', () => {
    const props = {
      intl: {
        formatMessage: jest.fn(),
      },
      addItem: jest.fn(),
      items: [
        { id: 1, label: 'item 1' },
        { id: 2, label: 'item 2' },
        { id: 3, label: 'item 3' },
      ],
    };
    const tree = createComponentWithIntl(<Page {...props} />).toJSON();
    expect(tree).toMatchSnapshot();
  })
})
