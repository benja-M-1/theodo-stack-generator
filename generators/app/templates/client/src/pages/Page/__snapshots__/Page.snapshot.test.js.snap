// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`The page component should renders correctly 1`] = `
<div
  className="page-container"
>
  <a
    onClick={[Function]}
    style={Object {}}
  >
    <span>
      Retour
    </span>
  </a>
  <p />
  <button
    onClick={[Function]}
  >
    <span>
      Click sur moi
    </span>
  </button>
  <div
    className="itemContainer"
  >
    <div>
      item 1
    </div>
    <div>
      item 2
    </div>
    <div>
      item 3
    </div>
  </div>
</div>
`;
